### PR TITLE
Switch from fnv to a identity hasher in extensions

### PR DESCRIPTION
**Pull request by mitsuhiko**
I think fnv is overkill for this `TypeId` is a u64 which is also what hashmaps use for hashing. So this should effectively become a noop.

### CODE REVIEW

File: src/extensions.rs [hunk 1]
```diff
@@ -1,11 +1,36 @@
 use std::any::{Any, TypeId};
 use std::collections::HashMap;
+use std::hash::{BuildHasherDefault, Hasher};
 use std::fmt;
 use std::hash::BuildHasherDefault;
```
Comment (DoumanAsh) [inline src/extensions.rs]: remove duplicate import of `BuildHasherDefault`

File: src/extensions.rs [hunk 2]
```diff
@@ -1,11 +1,36 @@
...
 
-use fnv::FnvHasher;
+pub(crate) struct IdHasher {
```
Comment (DoumanAsh) [inline src/extensions.rs]: Do we actually use `IdHasher` outside of this module? It seems unnecessary to put `pub(crate)` here

File: src/extensions.rs [hunk 3]
```diff
@@ -1,11 +1,36 @@
...
+    id: u64,
+}
+
+impl Default for IdHasher {
+    fn default() -> IdHasher {
+        IdHasher { id: 0 }
+    }
+}
+
+impl Hasher for IdHasher {
```
Comment (DoumanAsh) [inline src/extensions.rs:16]: Should we maybe just use default hasher from std lib?

**Review by DoumanAsh — COMMENTED** [review]
Removing superfluous dependency is always good idea!

**Review by fafhrd91 — APPROVED** [review]

**Pull request merged**